src/components/obb-clipping.js

```diff
@@ -147,28 +161,28 @@ AFRAME.registerComponent('obb-clipping', {
   },
 
   applyClippingPlanes: function (clipPlanes) {
-    // console.log('applyclipping', this.elementToClip);
-    // console.log('applyclipping', this);
     if (!this.elementToClip) {
       this.fetchElementToClip();
     }
     if (this.elementToClip) {
-      this.elementToClip.object3D.traverse((obj) => {
-        if (obj.type === 'Mesh') {
-          if (Array.isArray(obj.material)) {
-            obj.material.forEach((material) => {
-              material.clippingPlanes = clipPlanes;
-              material.clipIntersection = true;
-            });
-          } else {
-            obj.material.clippingPlanes = clipPlanes;
-            obj.material.clipIntersection = true;
-          }
-        }
-      });
+      this.applyClippingPlanesToObject(clipPlanes, this.elementToClip.object3D);
     }
   },
 
+  applyClippingPlanesToObject: function (clipPlanes, object3D) {
+    object3D.traverse((obj) => {
+      if (obj.type === 'Mesh') {
+        const materials = Array.isArray(obj.material)
+          ? obj.material
+          : [obj.material];
+        materials.forEach((material) => {
+          material.clippingPlanes = clipPlanes;
+          material.clipIntersection = true;
+        });
+      }
+    });
+  },
+
```

Comment:
If it's an hot path, you should avoid creating an array and prefer using for-of.
```js
   object3D.traverse((obj) => {
      if (obj.isMesh) {
        if (Array.isArray(obj.material)) {
          for (const material of obj.material) {
            material.clippingPlanes = clipPlanes;
            material.clipIntersection = true;
          }
        } else {
          obj.material.clippingPlanes = clipPlanes;
          obj.material.clipIntersection = true;
        } 
      }
    });
```

Reply:
thanks @vincentfretin, I just reverted back to this logic which was there before and i mistakingly tried to simplify